rule/receiver_naming.go

```diff
@@ -47,18 +47,63 @@ func (r *ReceiverNamingRule) configure(arguments lint.Arguments) {
 func (r *ReceiverNamingRule) Apply(file *lint.File, args lint.Arguments) []lint.Failure {
 	r.configureOnce.Do(func() { r.configure(args) })
 
+	typeReceiver := map[string]string{}
 	var failures []lint.Failure
+	for _, decl := range file.AST.Decls {
+		fn, ok := decl.(*ast.FuncDecl)
+		if !ok || fn.Recv == nil || len(fn.Recv.List) == 0 {
+			continue
+		}
 
-	fileAst := file.AST
-	walker := lintReceiverName{
-		onFailure: func(failure lint.Failure) {
-			failures = append(failures, failure)
-		},
-		typeReceiver:          map[string]string{},
-		receiverNameMaxLength: r.receiverNameMaxLength,
-	}
+		names := fn.Recv.List[0].Names
+		if len(names) < 1 {
+			continue
+		}
+		name := names[0].Name
+
+		if name == "_" {
+			failures = append(failures, lint.Failure{
+				Node:       decl,
+				Confidence: 1,
+				Category:   "naming",
+				Failure:    "receiver name should not be an underscore, omit the name if it is unused",
+			})
+			continue
+		}
+
+		if name == "this" || name == "self" {
+			failures = append(failures, lint.Failure{
+				Node:       decl,
+				Confidence: 1,
+				Category:   "naming",
+				Failure:    `receiver name should be a reflection of its identity; don't use generic names such as "this" or "self"`,
+			})
+			continue
+		}
```

Comment:
You could use a switch

You are  checking the same variable three times, also it would allow to remove the two continue, as they will be useless


```suggestion
		switch name {
			case "_":
				failures = append(failures, lint.Failure{
					Node:       decl,
					Confidence: 1,
					Category:   "naming",
					Failure:    "receiver name should not be an underscore, omit the name if it is unused",
				})
			case "this", "self":
				failures = append(failures, lint.Failure{
					Node:       decl,
					Confidence: 1,
					Category:   "naming",
					Failure:    `receiver name should be a reflection of its identity; don't use generic names such as "this" or "self"`,
				})
		}
```

Reply:
The switch on name will cope with the first two conditions for failure, the other conditions are not just on name thus IMO, adding a switch will not help in clarity (further, even with a switch `continue` is needed after adding a failure to do not evaluate the not-in-the-switch conditions)